packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -165,6 +179,30 @@ class Context {
     exitApp(-1);
   }
 
+  /// Determines if the platform is [TargetPlatform.ios] or [TargetPlatform.macos]
+  /// by checking the `PROJECT_DIR` or `PLATFORM_NAME`.
+  ///
+  /// If unable to determine the platform, defaults to [TargetPlatform.ios].
+  TargetPlatform getPlatform() {
```

Comment:
Did you consider updating `macos_assemble.sh` and `xcode_backend.sh` to pass in `--platform <macos or ios>` when they invoke `xcode_backend.dart`? I think that might be simpler, but let me know if I'm missing something